code/scripting/api/libs/graphics.cpp

```diff
@@ -1250,20 +1250,26 @@ static int drawString_sub(lua_State *L, bool use_resize_arg)
 		SCP_vector<int> linelengths;
 		SCP_vector<const char*> linestarts;
 
-		if (y2 >= 0 && y2 < y)
+		if (x2 <= x)
+		{
+			// This would pass a <=0 value to split_str
+			Warning(LOCATION, "Illegal values passed to gr.drawString: x2 (%d) must be greater than x (%d).", x2, x);
```

Comment:
Won't this throw a warning every time that swapped values are used?  Can we use a `static bool` to warn just once?

Reply:
Yeah, we could do that.  Done.